### PR TITLE
Kubeadm configfile beta - add documentation for images

### PR DESCRIPTION
**Pull request by fabriziopandini**
As requested in [this comment](https://github.com/kubernetes/community/pull/2451#issuecomment-410295978), this PR adds documentation about how to update / edit images linked from KEP0023 - Kubeadm configfile beta

/CC @timothysc @liztio

### CODE REVIEW

**Comment by rosti** [comment]
Perhaps `s/master/control-plane/g`?

File: keps/sig-cluster-lifecycle/0023-documentation-for-images.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+participant "user" as u 
+participant "kubeadm" as k
+participant "kubeadm-config\nConfigMap" as cm
+participant "kubeproxy-config\nConfigMap" as kpcm 
+participant "kubelet-config\nConfigMap-1.*+1" as kcm
+participant "kubelet" as kk
+participant "node\n(api object)" as n
+
+u->k: UpgradeConfiguration
+note over u, n:Upgrade configuration should allow only well-know changes to the cluster e.g. change of custom images if used
```
Comment (neolit123) [inline keps/sig-cluster-lifecycle/0023-documentation-for-images.md]: `well-know` -> `well known`

Comment (neolit123) [reply keps/sig-cluster-lifecycle/0023-documentation-for-images.md]: ` e.g. change` ->  `, e.g. the change`

File: keps/sig-cluster-lifecycle/0023-documentation-for-images.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+k->cm:read ClusterConfiguration
+cm->k:
+k->k:update\nClusterConfiguration\nusing api machinery
+k->cm:save updated ClusterConfiguration
+
+k->kpcm:read KubeProxy component config
+kpcm->k:
+k->k:update KubeProxy\ncomponent config\nusing api machinery
+k->kpcm:save updated KubeProxy component config
+note over kpcm, n:the updaed KubeProxy component config will\nbe used by the updated kube-proxy DaemonsSet
```
Comment (neolit123) [inline keps/sig-cluster-lifecycle/0023-documentation-for-images.md]: `DaemonsSet` -> `DaemonSet`

File: keps/sig-cluster-lifecycle/0023-documentation-for-images.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+participant "user" as u 
+participant "kubeadm" as k
+participant "kubelet" as kk
+participant "node\n(api object)" as n
+participant "kubeadm-config\nConfigMap" as cm
+participant "kubeproxy-config\nConfigMap" as kpcm 
+participant "kubelet-config\nConfigMap-1.*" as kcm
+
+u->k:provide\nInitConfiguration (with NodeRegistrationOptions, Master)\nClusterConfiguration\nKubeProxy component config\nKubelet component config
```
Comment (neolit123) [inline keps/sig-cluster-lifecycle/0023-documentation-for-images.md]: `Kubelet` -> `kubelet`
(case-sensitive search-and-replace would solve as there are more than one instances.)

File: keps/sig-cluster-lifecycle/0023-documentation-for-images.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+participant "kubeadm" as k
+participant "kubeadm-config\nConfigMap" as cm
+participant "node\n(api object)" as n
+
+
+u->k: 
+
+k->cm:read ClusterConfiguration
+cm->k:
+k->cm:remove Master\nfrom ClusterConfiguration.Status\n(only if the node is master)
```
Comment (neolit123) [inline keps/sig-cluster-lifecycle/0023-documentation-for-images.md]: `is master` -> `is a Control Plane node`

File: keps/sig-cluster-lifecycle/0023-documentation-for-images.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+
+k->kcm:read Kubelet component config
+kcm->k:
+k->k:update Kubelet\ncomponent config\nusing api machinery
+k->kcm:save updated Kubelet component config 
+k->kk:write kubelet component config\nto /var/lib/kubelet/config.yaml
+k->kk:write NodeRegistrationOptions\nto /var/lib/kubelet/kubeadm-flags.env
+kk->n:start node
+
+note over kcm, n:the updated Kubelet component config\nwill be used by other nodes\nby running\nkubeadm upgrade nodes locally
```
Comment (neolit123) [inline keps/sig-cluster-lifecycle/0023-documentation-for-images.md]: `\nby running` -> `\nwhen running`

File: keps/sig-cluster-lifecycle/0023-documentation-for-images.md [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+kk->n:start node
+
+note over kcm, n:the updated Kubelet component config\nwill be used by other nodes\nby running\nkubeadm upgrade nodes locally 
+
+```
+
+## kubeadm upgrade node
+
+```
+title kubeadm upgrade node (interactions with v1beta1 config)
```
Comment (neolit123) [inline keps/sig-cluster-lifecycle/0023-documentation-for-images.md]: `v1beta1 config` -> `the v1beta1 configuration`

File: keps/sig-cluster-lifecycle/0023-documentation-for-images.md [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+
+k->n:read kubeadm.alpha.kubernetes.io/cri-socket annotation 
+n->k:
+k->k:use CRIsocket\nto delete containers
+```
+
+## kubeadm upgrade
+
+```
+title kubeadm upgrade apply (interactions with v1beta1 config)
```
Comment (neolit123) [inline keps/sig-cluster-lifecycle/0023-documentation-for-images.md]: `v1beta1` -> `the v1beta1 configuration`

File: keps/sig-cluster-lifecycle/0023-documentation-for-images.md (hunk 2 above)
Comment (neolit123) [reply keps/sig-cluster-lifecycle/0023-documentation-for-images.md]: `updaed` -> `updated`

File: keps/sig-cluster-lifecycle/0023-documentation-for-images.md (hunk 3 above)
Comment (neolit123) [reply keps/sig-cluster-lifecycle/0023-documentation-for-images.md]: `component config` -> `component configuration`?
(present in other places too)

File: keps/sig-cluster-lifecycle/0023-documentation-for-images.md [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+kk->n:start node
+k->n:save NodeRegistrationOptions.CRISocket\nto kubeadm.alpha.kubernetes.io/cri-socket annotation 
+
+k->cm:add new Master\nto ClusterConfiguration.Status\n(only for join--master)
+```
+
+## kubeadm reset 
+
+```
+title kubeadm reset (interactions with v1beta1 config)
```
Comment (neolit123) [inline keps/sig-cluster-lifecycle/0023-documentation-for-images.md]: `v1beta1 config` -> `the v1beta1 configuration`

**Comment by neolit123** [comment]
thanks for the writeup @fabriziopandini :+1: 
please feel to ignore some of the comments if you think the current text is better, as is.

**Comment by liztio** [comment]
Thanks for this documentation @fabriziopandini! Super helpful to make this be more editable by other people.

**Pull request merged**